### PR TITLE
Add routing table retrieval and updating functions to Go API.

### PR DESCRIPTION
**Pull request by hodgestar**
This is needed by the routing table jsPlumb UI. We need at least a method for:
- Retrieving the routing table and channels, routing blocks and endpoints.
- Saving an updated routing table (and validating it).

Example JSON for routing table retrieval:

``` javascript
var routing = {
  campaign_id: 'campaign-1',
  channels: [{
    uuid: 'channel-uuid-1',
    tag: ['apposit_sms', '*121#'],
    name: '*121#',
    description: 'Apposit Sms: *121#',
    endpoints: [{uuid: 'endpoint-uuid-1', name: 'default'}]
  }],
  routing_blocks: [{
    uuid: 'routing-block-uuid-1',
    type: 'keyword',
    name: 'keyword-routing-block',
    description: 'Keyword',
    channel_endpoints: [{uuid: 'endpoint-uuid-2', name: 'default'}],
    conversation_endpoints: [{uuid: 'endpoint-uuid-3', name: 'default'}]
  }],
  conversations: [{
    uuid: 'routing-block-uuid-2',
    type: 'bulk-message',
    name: 'bulk-message-1',
    description: 'Some Bulk Message App',
    endpoints: [{uuid: 'endpoint-uuid-4', name: 'default'}]
  }],
  routing_entries: [{
    source: {uuid: 'endpoint-uuid-1'},
    target: {uuid: 'endpoint-uuid-2'}
  }]
};
```

### CODE REVIEW

**Comment by justinvdm** [comment]
@hodgestar We can have `campaign_id` outside of the `routing` object if you like, don't think it will affect the client side.

File: go/api/go_api/go_api.py [hunk 1]
```diff
@@ -76,12 +90,150 @@ def jsonrpc_campaigns(self):
...
+                    (e for r in routing_blocks
+                     for e in r['conversation_endpoints'])
+                )
+            )
+
+            return channel_endpoints, conversation_endpoints
+
+        def check_routing_table(endpoint_sets):
+            """Check that endpoints link from known channel (right) endpoints
+            to known conversation (left) endpoints or vice versa.
```
Comment (justinvdm) [inline go/api/go_api/go_api.py]: Shouldn't `right` and `left` be swapped around?

Comment (hodgestar) [inline go/api/go_api/go_api.py]: Yes, that's what my questions in #313 were about. Let me update this.

Comment (justinvdm) [inline go/api/go_api/go_api.py]: Ah ok cool, thought it may be related to that.

**Comment by justinvdm** [comment]
:+1:

**Comment by justinvdm** [comment]
Sorry for the lack of comments, couldn't find any faults to be honest. I do think I grokked the PR fairly well though.

**Pull request merged**